contrib/epee/include/net/net_helper.h

```diff
@@ -302,14 +336,17 @@ namespace net_utils
 				*/
 				boost::system::error_code ec;
 
-				size_t writen = m_socket.write_some(boost::asio::buffer(data, sz), ec);
+				size_t writen;
+				if(m_ssl)
+					writen = m_ssl_socket.write_some(boost::asio::buffer(data, sz), ec);
+				else
+					writen = m_socket.write_some(boost::asio::buffer(data, sz), ec);
```

Comment:
With all these, I'd have a socket pointer, which is set either to &m_socket or &m_ssl_socket, so you don't have to have a if(m_ssl) X; else Y; all over the place. Makes things simpler, and makes it less likely you forget one place.